src/workingtitle-vcockpits-instruments-navsystems/html_ui/Pages/VCockpit/Instruments/NavSystems/AS1000/PFD/AS1000_PFD.js

```diff
@@ -33,19 +33,65 @@ class AS1000_PFD extends BaseAS1000 {
         this.addEventLinkedPopupWindow(new NavSystemEventLinkedPopUpWindow("Procedures", "ProceduresWindow", new MFD_Procedures(), "PROC_Push"));
         this.addEventLinkedPopupWindow(new NavSystemEventLinkedPopUpWindow("CONFIG", "PfdConfWindow", new AS1000_PFD_ConfigMenu(), "CONF_MENU_Push"));
         this.maxUpdateBudget = 12;
-        let avionicsKnobIndex = 30;
-        let avionicsKnobValue = SimVar.GetSimVarValue("A:LIGHT POTENTIOMETER:" + this.avionicsKnobIndex, "number");
+        this.avionicsKnobIndex = 30;
+        this._cfgHandler = new ConfigLoader(this._xmlConfigPath);
+        this._pfdConfigDone = false;
+        this.pfdConfig();
     }
-    onUpdate(_deltaTime) {
-        let avionicsKnobValueNow = SimVar.GetSimVarValue("A:LIGHT POTENTIOMETER:" + this.avionicsKnobIndex, "number") * 100;
-        if (avionicsKnobValueNow != this.avionicsKnobValue) {
-            SimVar.SetSimVarValue("L:XMLVAR_AS1000_PFD_Brightness", "number", avionicsKnobValueNow);
-            SimVar.SetSimVarValue("L:XMLVAR_AS1000_MFD_Brightness", "number", avionicsKnobValueNow);
+
+    async pfdConfig() {
+        this.loadSavedBrightness("PFD");
+        this.loadSavedBrightness("MFD");
+        // It seems to be the case that in some planes the avionics brightness knob value changes
+        // as the plane is in the process of intializing -- the stock Bonanza seems to be one of
+        // these.  This causes the code in onUpdate that determines if the knob has moved from its
+        // prior value to trigger, meaning that the brightness that we set is immediately
+        // overridden by the new position of the knob.  The only way I have found to set this is,
+        // unfortunately, to stick a little pause in here to let the systems load a bit before
+        // completing the initialization of our current settings.  It's kinda gross, but since this
+        // is an async function it's not *too* bad, and the worst it means it that the hardware
+        // knob, if present, won't function for a few seconds.
+        //
+        // The 10000ms value here was experimentally derived by testing to see what caused loads in
+        // troublesome planes to succeed on my system.  It may need adjustment if other folks find
+        // themselves having that issue. 
+        await new Promise(resolve => setTimeout(resolve, 10000));
```

Comment:
Check if it would help subscribing to onFlightStart that is available in some base classes of the sim.
Or just check the use of the gamestates.

see BaseInstrument.js
```
    onFlightStart() {
        console.log("Flight Starting...");
        SimVar.SetSimVarValue("L:HUD_AP_SELECTED_SPEED", "Number", 0);
        SimVar.SetSimVarValue("L:HUD_AP_SELECTED_ALTITUDE", "Number", 0);
        this.dispatchEvent(new Event('FlightStart'));
    }
    onGameStateChanged(_oldState, _newState) {
        if (_newState != GameState.mainmenu) {
            this.createMainLoop();
            if (_oldState == GameState.loading && (_newState == GameState.ingame || _newState == GameState.briefing)) {
                this.reboot();
            }
            else if (_oldState == GameState.briefing && _newState == GameState.ingame) {
                this.onFlightStart();
            }
        }
        else {
            this.killMainLoop();
        }
        this._gameState = _newState;
    }
```

Reply:
That's a really great idea, and I've played with it a bit to try to make it work...  but I'm afraid it goes back to the same unreliable behavior I was trying to fix with that sleep.  :(   The problem is that we need to wait until the model interior xml has been loaded and parsed before we can assign a value to the avionics knob and _that_ needs to be done before we can flag the PFD ready for control.   But if we do this in onFlightStart we can't await the load there because it's not an async function.  So we go back to square one.   I'll keep thinking about other ways around this.

Reply:
This works now by waiting for a promise in onFlightStart